src/commands/publish.ts

```diff
@@ -489,14 +521,16 @@ export async function publishMain(argv: PublishOptions): Promise<any> {
     }
   }
 
+  let targetConfigList = config.targets || [];
+
   logger.info(`Looking for publish state file for ${newVersion}...`);
   const publishStateFile = `.craft-publish-${newVersion}.json`;
   const earlierStateExists = existsSync(publishStateFile);
   let publishState: PublishState;
   if (earlierStateExists) {
     logger.info(`Found publish state file, resuming from there...`);
     publishState = JSON.parse(readFileSync(publishStateFile).toString());
-    targetsToPublish = new Set(getAllTargetNames());
+    targetsToPublish = new Set(targetConfigList.map(getTargetId));
```

Comment:
This, combined with better splitting the `if` statements below is the real fix for the issue documented on the PR description.